examples/chainermn/cifar/train_cifar.py

```diff
@@ -0,0 +1,119 @@
+import argparse
+
+import chainermn
+import chainer
+import chainer.links as L
+from chainer import training
+from chainer.training import extensions
+
+from chainer.datasets import get_cifar10
+from chainer.datasets import get_cifar100
+
+import models.VGG
+
+
+def main():
+    parser = argparse.ArgumentParser(description='Chainer CIFAR example:')
+    parser.add_argument('--dataset', '-d', default='cifar10',
+                        help='The dataset to use: cifar10 or cifar100')
+    parser.add_argument('--batchsize', '-b', type=int, default=64,
+                        help='Number of images per GPU in a mini-batch')
+    parser.add_argument('--learnrate', '-l', type=float, default=0.05,
+                        help='Learning rate for SGD')
+    parser.add_argument('--epoch', '-e', type=int, default=300,
+                        help='Number of sweeps over the dataset to train')
+    parser.add_argument('--gpu', '-g', action='store_true',
+                        help='Use GPU (negative value indicates CPU)')
+    parser.add_argument('--out', '-o', default='result',
+                        help='Directory to output the result')
+    parser.add_argument('--resume', '-r', default='',
+                        help='Resume the training from snapshot')
+    parser.add_argument('--communicator', type=str, default='flat',
+                        help='Type of communicator')
+    args = parser.parse_args()
+
+    # Prepare ChainerMN communicator.
+    if args.gpu:
+        comm = chainermn.create_communicator(args.communicator)
+        device = comm.intra_rank
+    else:
+        comm = chainermn.create_communicator('naive')
+        device = -1
+
+    if comm.rank == 0:
+        print('==========================================')
+        print('Num process (COMM_WORLD): {}'.format(comm.size))
+        if args.gpu:
+            print('Using GPUs')
+        print('Using {} communicator'.format(args.communicator))
+        print('Num Minibatch-size: {}'.format(args.batchsize))
+        print('Num epoch: {}'.format(args.epoch))
+        print('==========================================')
+
+    # Set up a neural network to train.
+    # Classifier reports softmax cross entropy loss and accuracy at every
+    # iteration, which will be used by the PrintReport extension below.
+    if comm.rank == 0:
+        if args.dataset == 'cifar10':
+            print('Using CIFAR10 dataset.')
+            class_labels = 10
+            train, test = get_cifar10()
+        elif args.dataset == 'cifar100':
+            print('Using CIFAR100 dataset.')
+            class_labels = 100
+            train, test = get_cifar100()
+        else:
+            raise RuntimeError('Invalid dataset choice.')
+    model = L.Classifier(models.VGG.VGG(class_labels))
```

Comment:
There's a bug. `class_lables` is not assigned in processes with `rank > 0`.

Reply:
Change the turn of code 'train' before model.